sonic-net/sonic-buildimage
# Count the number of undeliverable IPinIP packets

#### Why I did it
When an IPinIP packet lands on a TOR where the inner destination IP is not learned, that packet gets dropped. Keep count of the number of dropped packets so we can monitor/detect any traffic issues.

##### Work item tracking
- Microsoft ADO **24106234**:

#### How I did it
Create a new table in COUNTERS_DB to store undeliverable IPinIP packets and update it whenever tunnel packet handler sniffs such a packet.

#### How to verify it
Send an IPinIP packet to a TOR where the inner destination IP of the packet is not resolved on the TOR. Verify that the `COUNTERS:IPINIP_TUNNEL` table roughly reflects the number of packets sent. We allow some inaccuracy in the counter due to poor scapy perfomance under load as being able to track trends is more important that a 100% accurate packet count.

Link to sonic-mgmt test: https://github.com/sonic-net/sonic-mgmt/pull/14128



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
